### PR TITLE
Add softmax as layer, so that the axis arg can be used.

### PR DESCRIPTION
**Pull request by TimZaman**
The softmax activation has an `axis` argument. However, it is impossible to use this arg while not creating a custom layer or custom activation. Such custom items does not promote portability of the model.

Here added is the softmax activation as a layer. In principle, I think all the activations could (or should) also be layers. The use-case for this one is straight forward, and there have been several questions on-line about this (all solved with custom layers).

The default axis arg for softmax is -1. This is valid when you are using `image_data_format` as `channels_last`, **but** when you are using `channels_first`, the softmax will be applied over the width axis (in the case of the common `NCHW` formatting).

### CODE REVIEW

File: keras/layers/advanced_activations.py [hunk 1]
```diff
@@ -208,3 +209,32 @@ def get_config(self):
         config = {'theta': float(self.theta)}
         base_config = super(ThresholdedReLU, self).get_config()
         return dict(list(base_config.items()) + list(config.items()))
+
+
+class SoftMax(Layer):
```
Comment (fchollet) [inline keras/layers/advanced_activations.py]: Just `Softmax` would be more consistent with other names I think.

Comment (TimZaman) [reply keras/layers/advanced_activations.py]: Agreed: changed.
